docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml

```diff
@@ -284,3 +303,73 @@
       environment:
         KUBECONFIG: "/etc/kubernetes/admin.conf"
       when: get_dm_default_registry_key.stdout == ""
+
+    - block:
+        - wait_for:
+            # waiting task after apply config to avoid failures getting info
+            timeout: 5
+            msg: Waiting after apply DM config
+
+        # If unlock task is triggered, is highly probable that config applied is ok. 
+        - name: Wait until unlock task triggered
+          shell: |
+            source /etc/platform/openrc;
+            system host-show "{{ get_host_name.stdout}}" | awk '$2 ~ /^task/ {print $4}'
+          register: get_show_task_status
+          until: ("Unlocking" in get_show_task_status.stdout)
+          retries: 800
+          delay: "{{wait_for_dm_unlock}}"
+          ignore_errors: yes
+
+        - name: Show waiting unlock
+          debug:
+            msg:
+            - "waiting: {{get_show_task_status.stdout}}"
+            - "waiting: {{get_show_task_status.stderr}}"
+            
+        - name: Retrieve kubectl resources reconciled status
+          shell: >-
+            kubectl -n deployment get datanetworks,hostprofiles,hosts,platformnetworks,systems,ptpinstances,ptpinterfaces |
+            awk '$NF ~ /^false/ {print}'
+          environment:
+            KUBECONFIG: "/etc/kubernetes/admin.conf"
+          register: get_recon_status_pre_unlock
+
+        - name: Show unreconciled resources
+          debug:
+            msg:
+            - "recon: {{get_recon_status_pre_unlock.stdout}}"
+
+        #get pod name and error logs
+        - name: Get dm pod name
+          shell: >-
+            kubectl -n platform-deployment-manager get pods |
+            awk '$1 ~ /^platform-deployment-manager/ {print $1}'
+          environment:
+            KUBECONFIG: "/etc/kubernetes/admin.conf"
+          register: get_dm_pod_name
+
+        - name: Retrieve dm logs
+          shell: >-
+            kubectl -n platform-deployment-manager logs "{{ get_dm_pod_name.stdout }}" |
+            awk '(/ERROR/ && !/validation/ && !/waiting/ && !(seen[$10, $NF]++)) {print}'
+          environment:
+            KUBECONFIG: "/etc/kubernetes/admin.conf"
+          register: get_logs_pre_unlock
+
+        - name: Show logs
+          debug:
+            msg:
+            - "Pod log: {{get_logs_pre_unlock.stdout}}"
+            - "err: {{get_logs_pre_unlock.stderr}}"
+
+        - name: Output dm logs
+          fail:
+            msg:
+              - "Fail waiting for host unlock to be triggered. It could be due to DM config errors"
+              - "UNRECONCILED resources: {{get_recon_status_pre_unlock.stdout}}"
+              - "{{get_logs_pre_unlock.stdout}}"
+          register: fail_dm_logs
+          when: ("Unlocking" not in get_show_task_status.stdout)
+
+      when: (deploy_config is defined and "subcloud" in get_region_name.stdout)
```

Comment:
The subcloud's region name is not reliable to check its dc role

Reply:
Done. Thanks Yuxing!